### DISCUSSION TITLE
What's the best way to remove default theme boxShadow

### DISCUSSION THREAD

**Discussion by arnaudjnn** [chakra-ui/chakra-ui#1921] [General] [answered]
What's the best way to remove default theme boxShadow on focus (light grey/blue)?

Would be awesome to remove it from the default theme, it's not really nice 😀

**Comment by ljosberinn** [comment]
that'd be on `theme.shadows.outline` for buttons at least. I couldn't find a lightgrey one just now but its probably on there too. If you override it with `none` that should remove it. It _is_ an accessibility feature though, see more context [here](https://github.com/WICG/focus-visible/issues/128#issuecomment-372069474).

**Comment by with-heart** [comment]
Following up on this, #2016 is a related issue and Segun posted a link to a solution in his reply.

**Comment by segunadebayo** [comment, marked as answer]
Hi @arnaudjnn,

Turning off all focus shadows might lead to an inaccessible experience, as users are not able to know the currently focused element on the screen.

I think the goal here might be to either change the styles of the box-shadow or remove it for non-keyboard interactions. 

Here are your options:
- To change the box-shadow, update the value of `theme.shadows.outline` to whatever feels like to you
- To remove it for non-keyboard interactions, all you need to do is to install the `focus-visible` package and set it up as mentioned here: https://github.com/chakra-ui/chakra-ui/blob/develop/packages/css-reset/README.md

Cheers

PS: That said, if you still want to go ahead to remove it, set `theme.shadows.outline`'s value to 'none'

**Reply by TheThirdRace** [reply]
I'd like to point the `focus-visible` package is causing some visual glitches.

The problem is that the box shadow isn't displayed when the focus is on some parts (mostly underlying layers) of dialogs (Modal, Drawer, etc.) and menus (Menu). The shadow is displayed correctly when the focus is on their children. This leads to a visual glitch where you can see the shadow being displayed and removed while moving focus within those components, which cannot be unseen after you see it ;)

You can achieve the same effect as with the `focus-visible` package, minus the glitches, by using this code snippet in your global styles:

```css
':focus:not(:focus-visible):not([role="dialog"]):not([role="menu"])': {
	boxShadow: 'none !important'
}
```

**Reply by paulbuechner** [reply]
#### Note - Box Shadow Bug 🐛 
If you experience issues relating to the `box-shadow` prop, when using `focus-visible` or the other solution @TheThirdRace pointed out, you might need to **overwrite** the global props passing **local** `css` props directly or via chakra `sx Prop` like this:

```html
<Button
    sx={{
      ":focus:not(:focus-visible)": {
        shadow: useColorModeValue(
          "0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05) !important",
          "rgba(0, 0, 0, 0.1) 0px 0px 0px 1px, rgba(0, 0, 0, 0.2) 0px 5px 10px, rgba(0, 0, 0, 0.4) 0px 15px 40px !important"
        ),
      },
    }}>
</Button>
```

The `!important` will overwrite the global properties set by focus-visible.

**Reply by RIP21** [reply]
@segunadebayo isn't it be great to include this answer somewhere in the docs? I think may encounter that and would want to get it improved at least (as Safar is still doesn't support that).

**Comment by brandiqa** [comment]
Here's the solution _for customizing outline focus color and other properties_ that I've settled with:

Step 1: First install `npm install --save focus-visible`. Make sure to import it in your root index file of your project like so:

```tsx
import "focus-visible/dist/focus-visible";
```

Step 2: Use component styling to customize the color and other properties of the outline focus using `_focusVisible` attribute. Here's an example for **Link** component:

```tsx
import type { ComponentStyleConfig } from "@chakra-ui/react";

const Link: ComponentStyleConfig = {
  baseStyle: {
    borderRadius: "lg",
    _focus: {
      boxShadow: "2px 2px 4px teal",
    },
    _focusVisible: {
      color: "gray.100",
      background: "gray.700",
      textDecoration:"underline",
    },
  },
};

export default Link;
```

You can also make these changes in the global theme without creating a custom component style, however, it will affect other HTML elements that receive focus. Hence the need for creating a custom component style.

Step 3: In your `theme.ts` code, turn off focus outline for mouse events like so. We'll also import our Link component with custom focus outline:

```tsx
import { extendTheme } from "@chakra-ui/react";
import Link from "./components/Link";

const theme = extendTheme({
  styles: {
    global: {      
      ".js-focus-visible :focus:not([data-focus-visible-added])": {
        outline: "none",
        boxShadow: "none",
      },
    },
  },
 components: {
    Link,
  },
});
export default theme;
```

Step 4: In your application root, inject your custom theme settings into `ChakraProvider` like so:

```tsx
import "focus-visible/dist/focus-visible";
import theme from "~/lib/theme";

export default function App() {
  return (
     <ChakraProvider theme={theme}>
        {...}
      </ChakraProvider>
  );
}
```
